### PR TITLE
feat(swaps): enable Base for swaps

### PR DESCRIPTION
**Pull request by nikoferro**
## **Description**

This PR enables Base for swaps. Related to this, it also upgrades `@metamask/swaps-controller` to `v8.0.0`. See `package.json` and `yarn.lock` changes

## **Related issues**

Depends on `@metamask/swaps-controller` to `v8.0.0`

## **Manual testing steps**

1. Enable swaps on base feature flag
2. Change network to base
3. Get a quote for a Swap

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by legobeat — DISMISSED** [review]

**Review by legobeat — DISMISSED** [review]

**Review by legobeat — DISMISSED** [review]
LGTM

**Comment by tommasini** [comment]
Hey @nikoferro, awesome work! I see dedupe issues, can you solve those, please?

File: app/store/sagas/index.ts [hunk 1]
```diff
@@ -15,6 +15,11 @@ import Engine from '../../core/Engine';
 import Logger from '../../util/Logger';
 import LockManagerService from '../../core/LockManagerService';
 import AppConstants from '../../../app/core/AppConstants';
+import { XMLHttpRequest as _XMLHttpRequest } from 'xhr2';
+
```
Comment (tommasini) [inline app/store/sagas/index.ts:19]: I don't understand why we need to check if XMLHttpRequest is undefined, could you elaborate why is this needed on the sagas file?

File: package.json [hunk 2]
```diff
@@ -337,6 +337,7 @@
     "uuid": "^8.3.2",
     "valid-url": "1.0.9",
     "vm-browserify": "1.1.2",
+    "xhr2": "^0.2.1",
```
[the remark this answers is not in the document]
Comment (nikoferro) [reply package.json]: done! 🎉

File: app/store/sagas/index.ts (hunk 1 above)
Comment (nikoferro) [reply app/store/sagas/index.ts:19]: hey @tommasini i think you saw this in slack, but for the sake of keeping the record here: As part of the upgrade, it seems we are removing (in v8) a no longer needed global polyfill for XMLHttpRequest. Everything was smooth until https://github.com/MetaMask/metamask-mobile/pull/9208 got merged, since it seems to rely on that same polyfill. This is a fix for this new issue generated by this upgrade

Comment (tommasini) [reply app/store/sagas/index.ts:19]: Thanks for the explanation! 
I'm trying to have more context on this and learn something from it

Because what I see that it's happening here it is that we are changing the XMLHttpRequest global variable of react native, that uses a polyfill that needs to work with IOS and Android with one [node.js based](https://www.npmjs.com/package/xhr2), I don't want to block this PR, but at the same time I don't feel confident that I'm know the enough to know if this change is okay

I looked into the RN source code and I can't understand why this would be undefined.

What gives me more confidence is to see that the XMLHttpRequest is only used internally on test/debug folders
image

**Comment by legobeat** [comment]
@nikoferro Could you run a `yarn deduplicate` and commit the results?

File: app/store/sagas/index.ts (hunk 1 above)
Comment (Gudahtt) [reply app/store/sagas/index.ts:19]: Hmm, this is confusing for me as well.

Could you help me understand how this relates to adding Base support, or to the swap controller update?

Comment (legobeat) [reply app/store/sagas/index.ts:19]: #9208 introduced test code with this inadvertent implicit dependency on the old version of the swaps controller. So those references need to be addressed for the swaps controller to be upgraded after that.

**Comment by legobeat** [comment]
> @nikoferro Could you run a `yarn deduplicate` and commit the results?

I went ahead and sorted this in 6f8d028e3a005f522456aaa8f546eba27d599314.

File: app/store/sagas/index.ts (hunk 1 above)
Comment (nikoferro) [reply app/store/sagas/index.ts:19]: @Gudahtt @legobeat `xhr2` was added in order to polyfill `XMLHttpRequest`. As @legobeat pointed out #9208 (an unrelated PR) depended on this. For some reason it seems the older `swaps-controller` version was exposing this as a global module. 

The `typeof` check was added because i cannot assign the value directly, since (probably due to a misconfiguration of the linter), the linter understands that `XMLHttpRequest` is already a global variable.

I could have started looking even deeper into the linter configuration on why this is happening but its leading to a never ending extending scope of this PR

**Review by legobeat — DISMISSED** [review]
LGTM.

Perhaps `xhr2` should actually be removed alongside a refactor of those unrelated tests in a separate PR.

**Review by legobeat — DISMISSED** [review]

**Review by legobeat — APPROVED** [review]

**Review by tommasini — APPROVED** [review]
Approving this PR now since the E2E are passing
We should indeed address this XMLHTMLREQUEST overwritten on a PR separated from this one
